AsSuperVisitor: type is not an erased subtype of supertype.

This is using checker-framework 3.0.1 with OpenJDK 11.0.2 with the NullnessChecker only.  I have the following simplified code in two classes, which compiles fine in plain Java:

```
package sup;

import java.util.function.Consumer;

public class Parent<T extends AutoCloseable>
{
    protected class Wrapper
    {
        protected Wrapper(Consumer<T> makeExpression)
        {
        }
    }
}
```

and

```
package sup;

import java.io.Closeable;

public class Child extends Parent<Closeable>
{
    Wrapper x = new Wrapper(Child::takesCloseable);

    private static void takesCloseable(Closeable rhs)
    {
    }
}
```

There is nothing special about Closeable here, it's just a handy interface from the core classes where Closeable extends AutoCloseable to shorten the example.  Some combination of the inner class in the parent using the parent's type parameter plus the method reference in the subclass seems to cause a problem.  The error goes through the well-named BugInCF class which is why I'm reporting it here.  Stack trace of errors is:

```
[ERROR] AsSuperVisitor: type is not an erased subtype of supertype.
[ERROR]   type: @UnknownKeyFor AutoCloseable
[ERROR]   superType: @UnknownKeyFor Closeable
[ERROR]   Compilation unit: C:\Users\neil\IdeaProjects\testchecker\mymodule\src\main\java\sup\Child.java
[ERROR]   Last visited tree at line 7 column 29:
[ERROR]       Wrapper x = new Wrapper(Child::takesCloseable);
[ERROR]   Exception: java.lang.Throwable; Stack trace: org.checkerframework.javacutil.BugInCF.<init>(BugInCF.java:25)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.errorTypeNotErasedSubtypeOfSuperType(AsSuperVisitor.java:154)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Declared(AsSuperVisitor.java:338)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Declared(AsSuperVisitor.java:28)
[ERROR]   org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:307)
[ERROR]   org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:57)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visit(AsSuperVisitor.java:85)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visitTypevar_NotTypevarNorWildcard(AsSuperVisitor.java:583)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visitTypevar_Declared(AsSuperVisitor.java:590)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visitTypevar_Declared(AsSuperVisitor.java:28)
[ERROR]   org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:631)
[ERROR]   org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:57)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visit(AsSuperVisitor.java:85)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.asSuper(AsSuperVisitor.java:67)
[ERROR]   org.checkerframework.framework.util.AnnotatedTypes.asSuper(AnnotatedTypes.java:112)
[ERROR]   org.checkerframework.framework.type.poly.AbstractQualifierPolymorphism$PolyCollector.visit(AbstractQualifierPolymorphism.java:451)
[ERROR]   org.checkerframework.framework.type.poly.AbstractQualifierPolymorphism$PolyCollector.visit(AbstractQualifierPolymorphism.java:401)
[ERROR]   org.checkerframework.framework.type.poly.AbstractQualifierPolymorphism$PolyCollector.access$100(AbstractQualifierPolymorphism.java:325)
[ERROR]   org.checkerframework.framework.type.poly.AbstractQualifierPolymorphism.resolve(AbstractQualifierPolymorphism.java:225)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.checkMethodReferenceAsOverride(BaseTypeVisitor.java:3032)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitMemberReference(BaseTypeVisitor.java:1619)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitMemberReference(BaseTypeVisitor.java:167)
[ERROR]   jdk.compiler/com.sun.tools.javac.tree.JCTree$JCMemberReference.accept(JCTree.java:2205)
[ERROR]   jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:88)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:301)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.checkArguments(BaseTypeVisitor.java:2824)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitNewClass(BaseTypeVisitor.java:1557)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitNewClass(BaseTypeVisitor.java:167)
[ERROR]   jdk.compiler/com.sun.tools.javac.tree.JCTree$JCNewClass.accept(JCTree.java:1711)
[ERROR]   jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:88)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:301)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:167)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.visitVariable(TreeScanner.java:223)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.visitVariable(SourceVisitor.java:100)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitVariable(BaseTypeVisitor.java:1043)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitVariable(BaseTypeVisitor.java:167)
[ERROR]   jdk.compiler/com.sun.tools.javac.tree.JCTree$JCVariableDecl.accept(JCTree.java:974)
[ERROR]   jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:88)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:301)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:167)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.visitClass(TreeScanner.java:187)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.visitClass(SourceVisitor.java:94)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.processClassTree(BaseTypeVisitor.java:384)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:339)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:167)
[ERROR]   jdk.compiler/com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:808)
[ERROR]   jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.visit(SourceVisitor.java:82)
[ERROR]   org.checkerframework.framework.source.SourceChecker.typeProcess(SourceChecker.java:1020)
[ERROR]   org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:523)
[ERROR]   org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:516)
[ERROR]   org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:182)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:828)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:132)
[ERROR]   jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1418)
[ERROR]   jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1375)
[ERROR]   jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:973)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:104)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:147)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:100)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:94)
[ERROR]   org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess(JavaxToolsCompiler.java:125)
[ERROR]   org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:169)
[ERROR]   org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:825)
[ERROR]   org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:129)
[ERROR]   org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]   org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR]   org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR]   org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR]   org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]   org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]   org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR]   org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR]   org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR]   org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR]   org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR]   org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
[ERROR]   org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
[ERROR]   org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
[ERROR]   java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]   java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]   java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]   java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]   org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR]   org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR]   org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR]   org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR]   org.codehaus.classworlds.Launcher.main(Launcher.java:47)
[ERROR] AsSuperVisitor: type is not an erased subtype of supertype.
[ERROR]   type: @Initialized @NonNull AutoCloseable
[ERROR]   superType: @Initialized @NonNull Closeable
[ERROR]   Compilation unit: C:\Users\neil\IdeaProjects\testchecker\mymodule\src\main\java\sup\Child.java
[ERROR]   Last visited tree at line 7 column 29:
[ERROR]       Wrapper x = new Wrapper(Child::takesCloseable);
[ERROR]   Exception: java.lang.Throwable; Stack trace: org.checkerframework.javacutil.BugInCF.<init>(BugInCF.java:25)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.errorTypeNotErasedSubtypeOfSuperType(AsSuperVisitor.java:154)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Declared(AsSuperVisitor.java:338)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visitDeclared_Declared(AsSuperVisitor.java:28)
[ERROR]   org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:307)
[ERROR]   org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:57)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visit(AsSuperVisitor.java:85)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visitTypevar_NotTypevarNorWildcard(AsSuperVisitor.java:583)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visitTypevar_Declared(AsSuperVisitor.java:590)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visitTypevar_Declared(AsSuperVisitor.java:28)
[ERROR]   org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:631)
[ERROR]   org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visit(AbstractAtmComboVisitor.java:57)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.visit(AsSuperVisitor.java:85)
[ERROR]   org.checkerframework.framework.type.AsSuperVisitor.asSuper(AsSuperVisitor.java:67)
[ERROR]   org.checkerframework.framework.util.AnnotatedTypes.asSuper(AnnotatedTypes.java:112)
[ERROR]   org.checkerframework.framework.type.poly.AbstractQualifierPolymorphism$PolyCollector.visit(AbstractQualifierPolymorphism.java:451)
[ERROR]   org.checkerframework.framework.type.poly.AbstractQualifierPolymorphism$PolyCollector.visit(AbstractQualifierPolymorphism.java:401)
[ERROR]   org.checkerframework.framework.type.poly.AbstractQualifierPolymorphism$PolyCollector.access$100(AbstractQualifierPolymorphism.java:325)
[ERROR]   org.checkerframework.framework.type.poly.AbstractQualifierPolymorphism.resolve(AbstractQualifierPolymorphism.java:225)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.checkMethodReferenceAsOverride(BaseTypeVisitor.java:3032)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitMemberReference(BaseTypeVisitor.java:1619)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitMemberReference(BaseTypeVisitor.java:167)
[ERROR]   jdk.compiler/com.sun.tools.javac.tree.JCTree$JCMemberReference.accept(JCTree.java:2205)
[ERROR]   jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:88)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:301)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.checkArguments(BaseTypeVisitor.java:2824)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitNewClass(BaseTypeVisitor.java:1557)
[ERROR]   org.checkerframework.checker.nullness.NullnessVisitor.visitNewClass(NullnessVisitor.java:553)
[ERROR]   org.checkerframework.checker.nullness.NullnessVisitor.visitNewClass(NullnessVisitor.java:54)
[ERROR]   jdk.compiler/com.sun.tools.javac.tree.JCTree$JCNewClass.accept(JCTree.java:1711)
[ERROR]   jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:88)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:301)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:167)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.visitVariable(TreeScanner.java:223)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.visitVariable(SourceVisitor.java:100)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitVariable(BaseTypeVisitor.java:1043)
[ERROR]   org.checkerframework.checker.initialization.InitializationVisitor.visitVariable(InitializationVisitor.java:167)
[ERROR]   org.checkerframework.checker.initialization.InitializationVisitor.visitVariable(InitializationVisitor.java:52)
[ERROR]   jdk.compiler/com.sun.tools.javac.tree.JCTree$JCVariableDecl.accept(JCTree.java:974)
[ERROR]   jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:88)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:301)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:167)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
[ERROR]   jdk.compiler/com.sun.source.util.TreeScanner.visitClass(TreeScanner.java:187)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.visitClass(SourceVisitor.java:94)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.processClassTree(BaseTypeVisitor.java:384)
[ERROR]   org.checkerframework.checker.initialization.InitializationVisitor.processClassTree(InitializationVisitor.java:295)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:339)
[ERROR]   org.checkerframework.common.basetype.BaseTypeVisitor.visitClass(BaseTypeVisitor.java:167)
[ERROR]   jdk.compiler/com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:808)
[ERROR]   jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
[ERROR]   org.checkerframework.framework.source.SourceVisitor.visit(SourceVisitor.java:82)
[ERROR]   org.checkerframework.framework.source.SourceChecker.typeProcess(SourceChecker.java:1020)
[ERROR]   org.checkerframework.common.basetype.BaseTypeChecker.typeProcess(BaseTypeChecker.java:523)
[ERROR]   org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:182)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:828)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:132)
[ERROR]   jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1418)
[ERROR]   jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1375)
[ERROR]   jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:973)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:104)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:147)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:100)
[ERROR]   jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:94)
[ERROR]   org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess(JavaxToolsCompiler.java:125)
[ERROR]   org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:169)
[ERROR]   org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:825)
[ERROR]   org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:129)
[ERROR]   org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]   org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR]   org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR]   org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR]   org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]   org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]   org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR]   org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR]   org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR]   org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR]   org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR]   org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
[ERROR]   org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
[ERROR]   org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
[ERROR]   java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]   java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]   java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]   java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]   org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR]   org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR]   org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR]   org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR]   org.codehaus.classworlds.Launcher.main(Launcher.java:47)
[ERROR] -> [Help 1]
```

--- a/checker/tests/nullness/Issue3022.java
+++ b/checker/tests/nullness/Issue3022.java
@@ -1,0 +1,11 @@
+abstract class Super3022<T> {
+    class Wrapper {
+        Wrapper(T key) {}
+    }
+}
+
+class Sub3022<K> extends Super3022<K> {
+    void wrap(K key) {
+        new Wrapper(key);
+    }
+}

--- a/framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java
+++ b/framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java
@@ -18,6 +18,7 @@ import com.sun.source.tree.NewArrayTree;
 import com.sun.source.tree.NewClassTree;
 import com.sun.source.tree.ReturnTree;
 import com.sun.source.tree.Tree;
+import com.sun.source.tree.Tree.Kind;
 import com.sun.source.tree.TypeCastTree;
 import com.sun.source.tree.VariableTree;
 import com.sun.source.util.TreePath;
@@ -1662,6 +1663,20 @@ public class AnnotatedTypeFactory implements AnnotationProvider {
                             "AnnotatedTypeFactory.getImplicitReceiver: enclosingClass()==null for element: "
                                     + element);
                 }
+                if (tree.getKind() == Kind.NEW_CLASS) {
+                    if (typeElt.getEnclosingElement() != null) {
+                        typeElt = ElementUtils.enclosingClass(typeElt.getEnclosingElement());
+                        if (typeElt == null) {
+                            // If the typeElt does not have an enclosing class, then the
+                            // NewClassTree does not have an implicit receiver.
+                            return null;
+                        }
+                    } else {
+                        // If the typeElt does not have an enclosing class, then the NewClassTree
+                        // does not have an implicit receiver.
+                        return null;
+                    }
+                }
                 // TODO: method receiver annotations on outer this
                 return getEnclosingType(typeElt, tree);
             }
@@ -2172,7 +2187,7 @@ public class AnnotatedTypeFactory implements AnnotationProvider {
         if (newClassTree.getEnclosingExpression() != null) {
             enclosingType = (AnnotatedDeclaredType) getReceiverType(newClassTree);
         } else {
-            enclosingType = null;
+            enclosingType = getImplicitReceiverType(newClassTree);
         }
         // Diamond trees that are not anonymous classes.
         if (TreeUtils.isDiamondTree(newClassTree) && newClassTree.getClassBody() == null) {

--- a/framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java
+++ b/framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java
@@ -1666,12 +1666,10 @@ public class AnnotatedTypeFactory implements AnnotationProvider {
                 if (tree.getKind() == Kind.NEW_CLASS) {
                     if (typeElt.getEnclosingElement() != null) {
                         typeElt = ElementUtils.enclosingClass(typeElt.getEnclosingElement());
-                        if (typeElt == null) {
-                            // If the typeElt does not have an enclosing class, then the
-                            // NewClassTree does not have an implicit receiver.
-                            return null;
-                        }
                     } else {
+                        typeElt = null;
+                    }
+                    if (typeElt == null) {
                         // If the typeElt does not have an enclosing class, then the NewClassTree
                         // does not have an implicit receiver.
                         return null;

--- a/framework/tests/all-systems/java8inference/Issue2975.java
+++ b/framework/tests/all-systems/java8inference/Issue2975.java
@@ -1,0 +1,22 @@
+package sup;
+
+import java.io.Closeable;
+import java.util.function.Consumer;
+
+class Issue2975<T extends AutoCloseable> {
+    static class Child extends Issue2975<Closeable> {
+        Wrapper y = new Wrapper(Child::takesCloseable);
+
+        private static void takesCloseable(Closeable rhs) {}
+    }
+
+    protected class Wrapper {
+        protected Wrapper() {}
+
+        protected Wrapper(Consumer<T> makeExpression) {}
+
+        protected Wrapper method(Consumer<T> makeExpression) {
+            throw new RuntimeException();
+        }
+    }
+}

--- a/framework/tests/all-systems/java8inference/Issue2975.java
+++ b/framework/tests/all-systems/java8inference/Issue2975.java
@@ -1,5 +1,3 @@
-package sup;
-
 import java.io.Closeable;
 import java.util.function.Consumer;
 

--- a/javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java
+++ b/javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java
@@ -794,6 +794,9 @@ public final class TreeUtils {
             default:
                 return null;
         }
+        if (receiver == null) {
+            return null;
+        }
 
         return TreeUtils.withoutParens(receiver);
     }

